Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -0,0 +1,300 @@
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using Mono.Cecil;
+
+using static RealmWeaver.Analytics;
+using static RealmWeaver.Metric;
+using OperatingSystem = RealmWeaver.Metric.OperatingSystem;
+
+namespace RealmWeaver
+{
+    internal static class AnalyticsUtils
+    {
+        public static string GetTargetOsName(FrameworkName frameworkName)
+        {
+            string targetOs = frameworkName.Identifier;
+
+            if (targetOs.ContainsIgnoreCase("android"))
+            {
+                return OperatingSystem.Android;
+            }
+
+            if (targetOs.ContainsIgnoreCase("ios"))
+            {
+                return OperatingSystem.Ios;
+            }
+
+            if (targetOs.ContainsIgnoreCase("mac"))
+            {
+                return OperatingSystem.MacOS;
+            }
+
+            if (targetOs.ContainsIgnoreCase("tvos"))
+            {
+                return OperatingSystem.TvOs;
+            }
+
+            if (targetOs.ContainsIgnoreCase("linux"))
+            {
+                return OperatingSystem.Linux;
+            }
+
+            if (targetOs.ContainsIgnoreCase("win") ||
+                targetOs.ContainsIgnoreCase("uap") ||
+                targetOs.ContainsIgnoreCase("win") ||
+                targetOs == ".NETFramework")
+            {
+                return OperatingSystem.Windows;
+            }
+
+            return Unknown(frameworkName.Identifier);
+        }
+
+        public static string SHA256Hash(byte[] bytes)
+        {
+            using var sha256 = SHA256.Create();
+            return BitConverter.ToString(sha256.ComputeHash(bytes));
+        }
+
+        public static string GetHostCpuArchitecture() => ConvertArchitectureToMetricsVersion(RuntimeInformation.OSArchitecture.ToString());
+
+        public static string GetTargetCpuArchitecture(ModuleDefinition module) => ConvertArchitectureToMetricsVersion(module.Architecture.ToString());
+
+        public static string GetHostOsName() =>
+            System.Environment.OSVersion.Platform switch
+            {
+                PlatformID.Win32NT or PlatformID.Win32S or PlatformID.Win32Windows or PlatformID.WinCE => OperatingSystem.Windows,
+                PlatformID.MacOSX => OperatingSystem.MacOS,
+                PlatformID.Unix => OperatingSystem.Linux,
+                _ => Unknown(System.Environment.OSVersion.Platform.ToString())
+            };
+
+        public static (string Name, string Version) GetFrameworkAndVersion(ModuleDefinition module, Config config)
+        {
+            if (config.TargetFramework.ContainsIgnoreCase("unity"))
+            {
+                return config.TargetFramework.ContainsIgnoreCase("editor") ?
+                    (Framework.UnityEditor, config.TargetFrameworkVersion) :
+                    (Framework.Unity, config.TargetFrameworkVersion);
+            }
```

Comment:
This is weird - should we just return `(config.TargetFramework, config.TargetFrameworkVersion)` here?

Reply:
Changes about this are made in the branch [ac/fixes-from-real-metrics-tests](https://github.com/realm/realm-dotnet/tree/ac/fixes-from-real-metrics-tests) for [PR 3244](https://github.com/realm/realm-dotnet/pull/3244)